### PR TITLE
Overview page on deployments

### LINKED ISSUE #2079: Create new page: Deploy your dbt project
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Deployment overview -- https://docs.getdbt.com/docs/dbt-cloud/deployments/deployment-overview

### What part(s) of the page would you like to see updated?

- Rename page to "About deployments" page. 
- Rewrite this page to be an dbt overview page for both Cloud and Core (which basically recommends Cloud as the best path but is being a good citizen about it so lists other options for the open source community).
- Cull (add) information from this page: https://docs.getdbt.com/docs/running-a-dbt-project/running-dbt-in-production
- Remove this ^ page after pulling relevant content. And create redirect to newly rewritten "About deployments" page

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by nghi-ly**
## Description & motivation
Rewrite About deployments page so it includes content from:

- https://docs.getdbt.com/docs/dbt-cloud/deployments/deployment-overview
- https://docs.getdbt.com/docs/running-a-dbt-project/running-dbt-in-production

closes #2079 

## Checklist
If you added new pages (delete if not applicable):
- [ ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`
- [ ] [Run link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/docs/deploy/deployment-overview.md [hunk 1]
```diff
@@ -3,16 +3,72 @@ title: "About deployments"
...
+* Accessing logs to help debug any issues.
+* Pulling the latest version of your git repo before running dbt (that is, continuous deployment).
+* Running your dbt project before merging code into master (that is, continuous integration).
+* Allowing access for team members that need to collaborate on your dbt project.
 
-In general, the dbt Cloud application deployment models fall into two categories: **Multi Tenant** and **Single Tenant**. These deployments are hosted on infrastructure managed by dbt Labs. Both models leverage AWS infrastructure as described in the [Architecture](deployment-architecture) section.
+## Run dbt in production
 
-For more information on these deployment models view the below pages.
+If you want to run dbt jobs on a schedule, you can use dbt Cloud, Airflow, Prefect, Dagster, automation server, or Cron.
```
Comment (joellabes) [inline website/docs/docs/deploy/deployment-overview.md]: Non-blocking: this reads as an exhaustive list. perhaps "tools such as dbt Cloud..."

File: website/docs/docs/deploy/deployment-overview.md [hunk 2]
```diff
@@ -3,16 +3,72 @@ title: "About deployments"
...
 
-- [Multi Tenant](multi-tenant-deployment)
-- [Single Tenant](single-tenant-deployment)
+## dbt Cloud
+
+We've built [dbt Cloud](https://www.getdbt.com/signup/) to empower data teams to easily run dbt in production. If you're interested in trying out dbt Cloud, you can [sign up for an account](https://cloud.getdbt.com/signup/).
```
Comment (joellabes) [inline website/docs/docs/deploy/deployment-overview.md]: Non-blocking: there are two variants of signup URL here. they both work, but a bit strange.

**Review by joellabes — APPROVED** [review]
lgtm!

File: website/docs/docs/build/custom-schemas.md [hunk 3]
```diff
@@ -158,7 +158,7 @@ Globally-scoped variables and variables defined on the command line with
 In the `generate_schema_name` macro examples shown above, the `target.name` context variable is used to change the schema name that dbt generates for models. If the `generate_schema_name` macro in your project uses the `target.name` context variable, you must additionally ensure that your different dbt environments are configured appropriately. While you can use any naming scheme you'd like, we typically recommend:
  - **dev**: Your local development environment; configured in a `profiles.yml` file on your computer.
  - **ci**:  A [continuous integration](cloud-enabling-continuous-integration-with-github) environment running on Pull Requests in GitHub, GitLab, etc.
- - **prod**: The production deployment of your dbt project, eg. in dbt Cloud, Airflow, or [similar](running-dbt-in-production)
+ - **prod**: The production deployment of your dbt project, eg. in dbt Cloud, Airflow, or [similar](/docs/deploy/deployments).
```
Comment (matthewshaver) [inline website/docs/docs/build/custom-schemas.md]:
```suggestion
 - **prod**: The production deployment of your dbt project, like in dbt Cloud, Airflow, or [similar](/docs/deploy/deployments).
```

File: website/docs/docs/deploy/deployment-overview.md [hunk 4]
```diff
@@ -3,16 +3,72 @@ title: "About deployments"
...
 
-:::info Deployments
+Running dbt in production means setting up a system to run a _dbt job on a schedule_, rather than running dbt commands manually from the command line. Your production dbt jobs should create the tables and <Term id="view">views</Term> that your business intelligence tools and end users query. Before continuing, make sure you understand dbt's approach to [managing environments](/docs/collaborate/environments).
 
-This section provides an overview of the deployment types available for dbt Cloud.
-If you’re interested in learning more about an Enterprise plan, contact us at sales@getdbt.com.
+In addition to setting up a schedule, there are other considerations when setting up dbt to run in production:
 
-:::
+* The complexity involved in creating a new dbt job, or editing an existing one.
```
Comment (matthewshaver) [inline website/docs/docs/deploy/deployment-overview.md]:
```suggestion
* The complexity involved in creating a new dbt job or editing an existing one.
```

File: website/docs/docs/deploy/deployment-overview.md [hunk 5]
```diff
@@ -3,16 +3,72 @@ title: "About deployments"
...
+* Setting up notifications if a step within your job returns an error code (for example, a model can't be built or a test fails).
+* Accessing logs to help debug any issues.
+* Pulling the latest version of your git repo before running dbt (that is, continuous deployment).
```
Comment (matthewshaver) [inline website/docs/docs/deploy/deployment-overview.md]:
```suggestion
* Pulling the latest version of your git repo before running dbt (continuous deployment).
```

File: website/docs/docs/deploy/deployment-overview.md [hunk 6]
```diff
@@ -3,16 +3,72 @@ title: "About deployments"
...
+* Running your dbt project before merging code into master (that is, continuous integration).
```
Comment (matthewshaver) [inline website/docs/docs/deploy/deployment-overview.md]:
```suggestion
* Running your dbt project before merging code into master (continuous integration).
```

**Pull request merged**